src/deluge/gui/ui/load/load_song_ui.cpp

```diff
@@ -228,9 +230,70 @@ ActionResult LoadSongUI::buttonAction(deluge::hid::Button b, bool on, bool inCar
 	return ActionResult::DEALT_WITH;
 }
 
+bool LoadSongUI::isLoadingSong() {
+	return getCurrentUI() == this || loadingSongInProgress;
+}
+
+void LoadSongUI::queueLoadNextSongIfAvailable(int8_t offset) {
+	if (!playbackHandler.isEitherClockActive()) {
+		// This method is intended just for queueing while playing
+		return;
+	}
+	if (loadingSongInProgress) {
+		return;
+	}
+	loadingSongInProgress = true;
+
+	outputTypeToLoad = OutputType::NONE;
+	currentDir.set(&currentSong->dirPath);
+
+	if (fileIndexSelected == -1) {
+		return;
```

Comment:
this needs to set loading song in progress to false before returning

It might be better to just move everything after the checks into another function so that this one only handles checking/setting the lock and then calling the actual queueNextSong function